### DISCUSSION TITLE
Companion web app that verifies the QR code?

### DISCUSSION THREAD

**Discussion by billylo1** [covidpass-org/covidpass#90] [Q&A] [answered]
HI, we are looking to adapt this solution for Ontario, Canada.  Our government only offers a signed PDF file, so I need to generate a QR code using the extracted data and offer a validation tool to confirm authenticity.

Has anyone started working on this type of feature?

thanks!

vaccine-ontario.ca (a volunteer group)
Billy.

**Comment by billylo1** [comment, marked as answer]
fyi... We implemented this idea at verifier.vaccine-ontario.ca. Works on iOS and Android browsers.

**Comment by dashed** [comment]
@billylo1 Is https://verifier.vaccine-ontario.ca/ open sourced anywhere?

**Reply by billylo1** [reply]
Coming soon...  The idea is not complex.  After the code on browser has checked the digital signature on the PDF, it generates a UUID and register that to the verifier server (no PII, just UUID, vaccineName, date, etc.) 

At verification time, check the data sent in the URL to see if they are consistent.

Hope this helps.

**Reply by marvinsxtr** [reply]
@dashed In Germany, we have the CovPass Check app, which should be able to verify all EU Digital COVID Certificates. See https://github.com/orgs/Digitaler-Impfnachweis
